virttest/storage.py

```diff
@@ -425,6 +455,36 @@ def auth_info_define_by_params(cls, image, params):
                 image, None, None, storage_type,
                 tls_creds=tls_creds, reconnect_delay=reconnect_delay
             ) if tls_creds or reconnect_delay else None
+        elif enable_curl:
+            # cookie data in a secure way, only for http/https
+            cookie_auth = cls(
+                image, params['curl_cookie_secret'],
+                params.get('curl_cookie_secret_format', 'raw'),
+                storage_type, data_type='cookie'
+            ) if (params['curl_protocol'] in ('http', 'https')
+                  and params.get('curl_cookie_secret')) else None
+
+            # sslverify, only for https/ftps
+            sslverify = params.get(
+                'curl_sslverify'
+            ) if params['curl_protocol'] in ('https', 'ftps') else None
+
+            # size of the read-ahead cache
+            readahead = params.get('curl_readahead')
+
+            # timeout for connections in seconds
+            timeout = params.get('curl_timeout')
+
+            # password
+            data = params.get('curl_passwd')
+            data_format = params.get('curl_passwd_format', 'raw')
+
+            auth = cls(
+                image, data, data_format, storage_type,
+                cookie_auth=cookie_auth, sslverify=sslverify,
+                readahead=readahead, timeout=timeout
+            ) if (data or cookie_auth or sslverify
+                  or readahead or timeout) else None
```

Comment:
```suggestion
            if any((data, cookie_auth, sslverify, readahead, timeout)):
               auth = cls(image, data, data_format, storage_type, 
                               cookie_auth=cookie_auth, sslverify=sslverify, 
                               readahead=readahead, timeout=timeout)
```
The `auth` has been set `None` in line 421 so you don't need to assign it again, I think.

Reply:
updated